modules/apps/27-interchain-accounts/host/keeper/relay.go

```diff
@@ -103,12 +100,6 @@ func (k Keeper) authenticateTx(ctx sdk.Context, msgs []sdk.Msg, connectionID, po
 		if !types.ContainsMsgType(allowMsgs, msg) {
 			return errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "message type not allowed: %s", sdk.MsgTypeURL(msg))
 		}
-
-		for _, signer := range msg.GetSigners() {
-			if interchainAccountAddr != signer.String() {
-				return errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "unexpected signer address: expected %s, got %s", interchainAccountAddr, signer.String())
-			}
-		}
```

Comment:
hahaha this is a very important check :sweat_smile: :money_with_wings:

Reply:
maybe @tac0turtle or @aaronc could explain how to get the signers correctly? I tried this:
```diff
+               signers, _, err := k.cdc.GetMsgV1Signers(msg)
+               if err != nil {
+                       return err
+               }
+
+               for _, signer := range signers {
+                       if interchainAccountAddr != string(signer) {
+                               return errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "unexpected signer address: expected %s, got %s", interchainAccountAddr, string(signer))
+                       }
+               }
+
```

But the signer doesn't match up, tests fail

Reply:
This works for now:
```diff
+               signers, _, err := k.cdc.GetMsgV1Signers(msg)
+               if err != nil {
+                       return err
+               }
+
+               for _, signer := range signers {
+                       if interchainAccountAddr != sdk.AccAddress(signer).String() {
+                               return errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "unexpected signer address: expected %s, got %s", interchainAccountAddr, sdk.AccAddress(signer).String())
+                       }
+               }
+
        }
```

We store the int

Reply:
https://github.com/cosmos/ibc-go/commit/838c3b1c2c19773deb906a393456717c4d506c8b

Reply:
🪦